integration/src/main/java/scripts/testscripts/DatasetPermissionOperations.java

```diff
@@ -0,0 +1,210 @@
+    CreateDatasetRequest request = datasetRequestForSnapshot(userTestSnapshotId);
+    var datasetId = adminCreateDataset(request);
+
+    // But admin cannot access the underlying preview data
+    assertThrows(Exception.class, () -> adminDatasetsApi.listDatasetPreviewTables(datasetId));
+    assertThat(
+        adminDatasetsApi.getApiClient().getStatusCode(),
+        is(HttpStatusCodes.STATUS_CODE_UNAUTHORIZED));
+  }
+
+  private void setTestSnapshotPermissionForRegularUser(String policy) throws Exception {
+    // first clear the shared snapshot of policy state caused by previous tests
+    clearTestSnapshotPermissions();
+    adminSnapshotsApi.addSnapshotPolicyMember(
+        adminTestSnapshotId, policy, new PolicyMemberRequest().email(regularUser.userEmail));
+  }
+
+  private void clearTestSnapshotPermissions() throws Exception {
+    for (String policy : List.of("reader", "discoverer")) {
+      adminSnapshotsApi.deleteSnapshotPolicyMember(
+          adminTestSnapshotId, policy, regularUser.userEmail);
+    }
+  }
+
+  private CreateDatasetRequest datasetRequestForSnapshot(UUID snapshotId) {
+    final String metadata = """
+        {"name": "test"}""";
+    return new CreateDatasetRequest()
+        .catalogEntry(metadata)
+        .storageSourceId(snapshotId.toString())
+        .storageSystem(StorageSystem.TDR);
+  }
+
+  private UUID adminCreateDataset(CreateDatasetRequest request) throws Exception {
+    var datasetId = adminDatasetsApi.createDataset(request).getId();
+    assertThat(adminDatasetsApi.getApiClient().getStatusCode(), is(HttpStatusCodes.STATUS_CODE_OK));
+    datasetIds.add(datasetId);
+    return datasetId;
+  }
+
+  private UUID createSnapshot(SnapshotsApi snapshotsApi) throws Exception {
+    assertNotNull(defaultProfileId);
+    var request =
+        new SnapshotRequestModel()
+            .name("catalog_integration_test_" + System.currentTimeMillis())
+            .description("catalog test snapshot")
+            .profileId(defaultProfileId)
+            .addContentsItem(
+                new SnapshotRequestContentsModel()
+                    .datasetName(TEST_DATASET_NAME)
+                    .mode(SnapshotRequestContentsModel.ModeEnum.BYFULLVIEW));
+    UUID snapshotId = ApiHelpers.synchronousCreateSnapshot(snapshotsApi, request);
+    log.info("created snapshot " + snapshotId);
+    snapshotIds.add(snapshotId);
+    return snapshotId;
+  }
+
+  @Override
+  public void cleanup(List<TestUserSpecification> testUsers) throws Exception {
+    for (var datasetId : datasetIds) {
+      adminDatasetsApi.deleteDataset(datasetId);
+      log.info("deleted dataset " + datasetId);
+    }
+    for (var snapshotId : snapshotIds) {
+      try {
+        ApiHelpers.synchronousDeleteSnapshot(adminSnapshotsApi, snapshotId);
+      } catch (Exception e) {
+        ApiHelpers.synchronousDeleteSnapshot(userSnapshotsApi, snapshotId);
+      } finally {
+        log.info("deleted snapshot " + snapshotId);
+      }
+    }
```

Comment:
Now that there are only two snapshots, `adminTestSnapshotId` and `userTestSnapshotId`, we no longer need `snapshotIds` and this loop.
```suggestion
    ApiHelpers.synchronousDeleteSnapshot(adminSnapshotsApi, adminTestSnapshotId);
    log.info("deleted snapshot " + adminTestSnapshotId);
    ApiHelpers.synchronousDeleteSnapshot(userSnapshotsApi, userTestSnapshotId);
    log.info("deleted snapshot " + userTestSnapshotId);
```

We could remove the `log`s too; they were nice when the tests were first being debugged but are probably less useful now. Since we're always using the same dataset for all tests, it wouldn't be hard to find the snapshots from that dataset if we needed to.